app/models/user.rb

```diff
@@ -54,25 +62,28 @@ def set_default_role
     self.role ||= :user
   end
 
-  def confirm_piv_cac?(proposed_uuid)
-    x509_dn_uuid == proposed_uuid if proposed_uuid
+  def piv_cac_configuration
+    PivCacConfiguration.new(self)
   end
 
-  def piv_cac_enabled?
-    PivCacLoginOptionPolicy.new(self).enabled?
+  def auth_app_configuration
+    AuthAppConfiguration.new(self)
   end
 
-  def piv_cac_available?
-    PivCacLoginOptionPolicy.new(self).available?
+  def personal_key_configuration
+    PersonalKeyConfiguration.new(self)
   end
 
   def need_two_factor_authentication?(_request)
     two_factor_enabled?
   end
 
   def two_factor_enabled?
-    phone_configurations.any?(&:mfa_enabled?) || totp_enabled? || piv_cac_enabled? ||
-      webauthn_enabled?
+    two_factor_configurations.any?(&:mfa_enabled?)
```

Comment:
Could we move this out of the User model and into a dedicated Policy Object?

Reply:
I'm considering creating a two factor method concern that could bundle all of the mfa stuff together.